src/config/index.js

```diff
@@ -1,92 +1,518 @@
 const fs = require("fs");
-const Path = require("path");
 const YAML = require("yaml");
 const Schema = require("./schema");
 
-function Config({env, configFile}) {
-  if (!fs.existsSync(configFile)) {
-    throw new Error(`THE RESTQA CONFIG FILE IS MISSING (${configFile})`);
+class Config {
+  constructor() {
+    this._config = null || {};
   }
 
-  const envVar = {
-    identify: (value) => value instanceof RegExp,
-    tag: "!env-var",
-    resolve(doc, node) {
-      const {strValue} = node;
-      return (
-        process.env[strValue] ||
-        `${strValue} not found in the environment variable`
+  load(filename) {
+    if (!fs.existsSync(filename)) {
+      throw new ReferenceError(
+        `The configuration file located at "${filename}" doesn't exist.`
       );
     }
-  };
+    const file = fs.readFileSync(filename, "utf8");
+    const envVar = {
+      identify: (value) => value instanceof RegExp,
+      tag: "!env-var",
+      resolve(doc, node) {
+        const {strValue} = node;
+        return (
+          process.env[strValue] ||
+          `${strValue} not found in the environment variable`
+        );
+      }
+    };
 
-  const file = fs.readFileSync(configFile, "utf8");
-  const config = YAML.parse(file, {customTags: [envVar]});
-  const envs = config.environments.map((env) => env.name);
+    const config = YAML.parse(file, {customTags: [envVar]});
+    Schema.validate(config);
+    this._config = config;
+    this._config.tests.unit = new UnitTest(this._config.tests.unit);
+    this._config.tests.integrations = (
+      this._config.tests.integrations || []
+    ).map((_) => new IntegrationTest(_));
+    this._config.tests.performance = new PerformanceTest(
+      this._config.tests.performance
+    );
+    this._config.specification = new Specification(this._config.specification);
+    this._config.collection = new Collection(this._config.collection);
+    this._config.plugins = (this._config.plugins || []).map(
+      (_) => new Plugin(_)
+    );
+    this._config.settings = new Settings(this._config.settings);
+  }
+
+  getCode() {
+    return this._config.metadata.code;
+  }
```

Comment:
Maybe what are your looking for is something like:
```js
get code() {
    return this._config.metadata.code;
}
```

Elsewhere:
```js
config.code;

// instead of

config.getCode();
```

Reply:
We could tackle this in another PR.

Reply:
👍